### DISCUSSION TITLE
[1.1.435] Depth Anything

### DISCUSSION THREAD

**Discussion by huchenlei** [Mikubill/sd-webui-controlnet#2549] [General]
## Depth Anything project
https://github.com/LiheYoung/Depth-Anything

## How to use
Download depth_anything ControlNet model [here](https://huggingface.co/spaces/LiheYoung/Depth-Anything/resolve/main/checkpoints_controlnet/diffusion_pytorch_model.safetensors). You need to rename the file for ControlNet extension to correctly recognize it. We recommend user to rename it as `control_sd15_depth_anything`.

![1705956857555](https://github.com/Mikubill/sd-webui-controlnet/assets/20929282/a8534fa3-5977-49ba-b2dd-bf41fada847e)

## Model comparison
Depth anything comes with a preprocessor and a new SD1.5 ControlNet model trained with images annotated by this preprocessor. They do not need to be used together. Here is a grid of result comparisons against existing preprocessors/models.

input image
![input](https://github.com/Mikubill/sd-webui-controlnet/assets/20929282/c3fd676a-34c4-4d62-8b14-6f8906710b72)

midas
![midas](https://github.com/Mikubill/sd-webui-controlnet/assets/20929282/432b93d5-b7f9-4ec8-89f4-98e8c47ba730)

leres
![leres](https://github.com/Mikubill/sd-webui-controlnet/assets/20929282/b289541f-2f55-44f0-a5e2-96e5ea114cdc)

leres++
![leres++](https://github.com/Mikubill/sd-webui-controlnet/assets/20929282/c7c02f25-9b47-47b0-bd09-2a90373b69f7)

depth_anything
![depth_anything](https://github.com/Mikubill/sd-webui-controlnet/assets/20929282/4c56b235-d965-4634-bb89-2dbef7c475ab)

xyz grid
![xyz_grid-0000-4139139971](https://github.com/Mikubill/sd-webui-controlnet/assets/20929282/d3683d38-7d45-4334-9e35-68aef8e8422a)

## Conslusion
I would say depth_anything might not outperforms depth_leres++ for amount details captured. We still adds the preprocessor anyway as it is fairly lightweight code wise.

## Update 2024-01-23
According to thread https://www.reddit.com/r/StableDiffusion/comments/19d7fx1/controlnet_has_been_updated_to_1435_includes/, depth_anything is better at handling motion blur and has better temporal consistency/stability. Maybe it is a superior choice when doing video generation, i.e. AnimateDiff.

**Comment by aimarxjg** [comment]
@huchenlei大佬，求助，我已经下载了1.5G的深度模型改名后放入了F:\sd-webui-aki-v4.5\extensions\sd-webui-controlnet\models 文件夹里面
![QQ图片20240123131414](https://github.com/Mikubill/sd-webui-controlnet/assets/132532636/a67865a7-46c5-4866-a24b-f0799746ab8d)

在文生图下使用CN 点击预览就错误
![555714](https://github.com/Mikubill/sd-webui-controlnet/assets/132532636/df89bfd9-b9b2-48cd-aa6e-caed72f62331)


开始进入WEBUI控制台就提示错误
![QQ图片20240123131003](https://github.com/Mikubill/sd-webui-controlnet/assets/132532636/66b3d041-3c65-4ac0-918e-225ab0487b76)

点击预览报错 后台显示
![QQ图片20240123131121](https://github.com/Mikubill/sd-webui-controlnet/assets/132532636/49bc4ba0-f29f-46ea-8fb4-16605fd0226f)

今天更新这个后我发现修手也报错了，还是一样的提示
![66](https://github.com/Mikubill/sd-webui-controlnet/assets/132532636/3ca3c370-aa5c-4b77-aaba-2f9039a83003)
![77](https://github.com/Mikubill/sd-webui-controlnet/assets/132532636/8a1d4844-3f49-48f9-ae8f-c3763af6c375)
![885](https://github.com/Mikubill/sd-webui-controlnet/assets/132532636/d34fa5e6-220e-4bb2-abc6-4e18d4af476a)

**Reply by huchenlei** [reply]
gfw的原因国内下载github依赖会有问题。我看到秋叶交了用环境变量换wheel url的pr你可以去问问他换源的事情。

**Reply by aimarxjg** [reply]
好的，谢谢~

**Reply by softhearter** [reply]
我也遇到相同的问题，我发现在如下图的scripts中的logging和文件logging_名称重复
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/101276167/c0efdd17-6ea2-4104-82f4-a508a6bb7949)
会导致一下的问题
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/101276167/123443c1-1aea-425b-9290-16d6c053a3a9)
应为我们的环境做了处理将修手的文件直接加入文件会导致import的时候定位不到需要使用的hand_refiner包也会导致该错误

**Reply by huchenlei** [reply]
logging是python标准包的一部分，local的文件是不应该shadow标准包的namespace的

**Reply by softhearter** [reply]
我比较疑问的是这个logging.py这样写不会有问题吗我之前一直在他程序引用的时候出现错误直到我把这个scripts中间的logging.py改成logging_.py并修改了相关引用才让这个手部修复变得正常，请问有什么更好的办法解决吗，感谢你的回复

**Reply by huchenlei** [reply]
这里root是scripts。logging是scripts module下的namespace，不会和global的logging冲突。

**Reply by softhearter** [reply]
根据你的回答我认为应该是我把handfiner给搞错地方了，由于某些不可抗因素我没有办法从ui界面直接下载到拓展包，我想请教您一个问题
![a4c9ac92bf65891a9abe19eed447873](https://github.com/Mikubill/sd-webui-controlnet/assets/101276167/64c0a757-d25a-4d3d-bc94-4c7104070735)
这个包他本来应该是放在那个目录下的，我的报错基本上来自于上面的包的安放位置不确定导致的，于是我修改的logging，感谢你的回复

ps;我修改过logging.py为logging_.py后可以正常运行hand refiner，但是您上面提到的确实是对的，那么问题应该是我存放package的位置存在问题，所以我想问问这个包它本来应该到那里去 QAQ

**Reply by huchenlei** [reply]
这个包是要pip安装到A1111的venv里的。你把代码checkout下来，activate A1111的venv，然后pip install {本地的repo路径}。

**Reply by softhearter** [reply]
好的，感谢你

**Comment by Cokeheart** [comment]
我也是这问题，我的handrefiner安装正常了，但是depth还是不能正常安装，这是为何
Commit hash: cf2772fab0af5573da775e7437e6acdca424f26e
Installing sd-webui-controlnet requirement: depth_anything
Couldn't install sd-webui-controlnet requirement: depth_anything.
Command: "D:\sd-webui-aki-v4.6\python\python.exe" -m pip install https://modelscope.cn/api/v1/studio/hanamizukiai/resolver/gradio/wheel/depth_anything-2024.1.22.0-py2.py3-none-any.whl --prefer-binary
Error code: 1
stdout: Looking in indexes: https://mirror.baidu.com/pypi/simple
Looking in links: https://mirror.sjtu.edu.cn/pytorch-wheels/torch_stable.html, https://mirrors.aliyun.com/pytorch-wheels/torch_stable.html, https://www.modelscope.cn/api/v1/studio/hanamizukiai/resolver/gradio/pypi-index/index.html
Collecting depth-anything==2024.1.22.0

stderr:   ERROR: HTTP error 404 while getting https://modelscope.cn/api/v1/studio/hanamizukiai/resolver/gradio/wheel/depth_anything-2024.1.22.0-py2.py3-none-any.whl
ERROR: Could not install requirement depth-anything==2024.1.22.0 from https://modelscope.cn/api/v1/studio/hanamizukiai/resolver/gradio/wheel/depth_anything-2024.1.22.0-py2.py3-none-any.whl because of HTTP error 404 Client Error: Not Found for url: https://modelscope.cn/api/v1/studio/hanamizukiai/resolver/gradio/wheel/depth_anything-2024.1.22.0-py2.py3-none-any.whl for URL https://modelscope.cn/api/v1/studio/hanamizukiai/resolver/gradio/wheel/depth_anything-2024.1.22.0-py2.py3-none-any.whl

Warning: Failed to install depth_anything. Some processors will not work.
